File: ta4j-core/src/main/java/org/ta4j/core/num/NumFactoryProvider.java
package org.ta4j.core.num;

/**
 * @author Lukáš Kvídera
 */
public final class NumFactoryProvider {
  private static NumFactory defaultNumFactory = DoubleNumFactory.getInstance();


  private NumFactoryProvider() {
    // utility class
  }


  public static NumFactory getDefaultNumFactory() {
    return defaultNumFactory;
  }


  public static void setDefaultNumFactory(final NumFactory numFactory) {
    defaultNumFactory = numFactory;
  }
}

File: ta4j-core/src/main/java/org/ta4j/core/num/NumFactoryProvider.kt
package org.ta4j.core.num;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.UtilityClass;

/**
 * @author Lukáš Kvídera
 */
@UtilityClass
public final class NumFactoryProvider {
  @Getter
  @Setter
  private static NumFactory defaultNumFactory = DoubleNumFactory.getInstance();
}
